resources/fetchers/kube_fetcher_test.go

```diff
@@ -0,0 +1,237 @@
+	switch items := items.(type) {
+	case []v1.Pod:
+		return PtrMap(items)
+	case []rbacv1.Role:
+		return PtrMap(items)
+	default:
+		return nil
+	}
+}
+
+func PtrMap[In any](items []In) []interface{} {
+	vsm := make([]interface{}, len(items))
+	for i := range items {
+		vsm[i] = &items[i]
+	}
+	return vsm
+}
+
+func Map[In fetching.Resource](resources []In) []interface{} {
+	vsm := make([]interface{}, len(resources))
+	for i, v := range resources {
+		vsm[i] = v.GetData()
+	}
+	return vsm
+}
+
+func (s *KubeFetcherTestSuite) TestKubeFetcher_TestFetch() {
+	empty := v1.PodList{}
+	pod := v1.PodList{Items: []v1.Pod{{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-pod",
+			Namespace: "kube-system",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:            "nginx",
+					Image:           "nginx",
+					ImagePullPolicy: "Always",
+				},
+			},
+		},
+	}}}
+	pods := v1.PodList{Items: []v1.Pod{
+		{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-pod",
+				Namespace: "kube-system",
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:            "nginx",
+						Image:           "nginx",
+						ImagePullPolicy: "Always",
+					},
+				},
+			},
+		},
+		{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Pod",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-pod2",
+				Namespace: "kube-system",
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:            "nginx",
+						Image:           "nginx",
+						ImagePullPolicy: "Always",
+					},
+				},
+			},
+		},
+	}}
+	roles := rbacv1.RoleList{Items: []rbacv1.Role{
+		{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Role",
+				APIVersion: "rbac.authorization.k8s.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-role1",
+				Namespace: "default",
+			},
+			Rules: []rbacv1.PolicyRule{
+				{Verbs: []string{"get"}},
+			},
+		},
+		{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Role",
+				APIVersion: "rbac.authorization.k8s.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-role2",
+				Namespace: "default",
+			},
+			Rules: []rbacv1.PolicyRule{
+				{Verbs: []string{"list"}},
+			},
+		},
+		{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Role",
+				APIVersion: "rbac.authorization.k8s.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-role3",
+				Namespace: "default",
+			},
+			Rules: []rbacv1.PolicyRule{
+				{Verbs: []string{"create"}},
+			},
+		},
+	}}
+	tests := []runtime.Object{
+		&empty,
+		&pod,
+		&pods,
+		&roles,
+	}
+
+	for i, tt := range tests {
+		s.Run(fmt.Sprintf("Kube api test %v", i), func() {
+			client := k8sfake.NewSimpleClientset(tt)
+			provider := MockProvider(client)
```

Comment:
Can be part of the suite if you want to.
can utilize SetupTest and TearDownTest for this purpose.